src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -412,59 +412,27 @@ public abstract class SendReceiveTest : ClientWebSocketTestBase
-                    Assert.Equal(
-                        ResourceHelper.GetExceptionMessage("net_WebSockets_InvalidState", "Aborted", "Open, CloseSent"),
-                        newReceiveException.Message);
+                Assert.Equal(WebSocketError.ConnectionClosedPrematurely, pendingReceiveException.WebSocketErrorCode);
 
-                    Assert.Equal(WebSocketState.Aborted, clientSocket.State);
-                    Assert.Null(clientSocket.CloseStatus);
-                }
-                else if (PlatformDetection.IsUap)
+                if (PlatformDetection.IsUap)
                 {
-                    _output.WriteLine("WinRTWebSocket-based implementation.");
-
                     const uint WININET_E_CONNECTION_ABORTED = 0x80072EFE;
 
-                    WebSocketException pendingReceiveException = await Assert.ThrowsAsync<WebSocketException>(() => pendingReceiveAsync);
-                    Assert.Equal(WebSocketError.ConnectionClosedPrematurely, pendingReceiveException.WebSocketErrorCode);
                     Assert.NotNull(pendingReceiveException.InnerException);
                     Assert.Equal(WININET_E_CONNECTION_ABORTED, (uint)pendingReceiveException.InnerException.HResult);
-
-                    WebSocketException newReceiveException =
-                        await Assert.ThrowsAsync<WebSocketException>(() => ReceiveAsync(clientSocket, recvSegment, cts.Token));
-                    Assert.Equal(WebSocketError.Success, newReceiveException.WebSocketErrorCode);
-                    Assert.Equal(
-                        ResourceHelper.GetExceptionMessage("net_WebSockets_InvalidState", "Aborted", "Open, CloseSent"),
-                        newReceiveException.Message);
-
-                    Assert.Equal(WebSocketState.Aborted, clientSocket.State);
-                    Assert.Null(clientSocket.CloseStatus);
                 }
-                else
-                {
-                    _output.WriteLine("[Non-Windows] ManagedWebSocket-based implementation.");
 
-                    WebSocketException pendingReceiveException = await Assert.ThrowsAsync<WebSocketException>(() => pendingReceiveAsync);
-                    Assert.Equal(WebSocketError.ConnectionClosedPrematurely, pendingReceiveException.WebSocketErrorCode);
-
-                    WebSocketException newReceiveException =
+                WebSocketException newReceiveException =
                         await Assert.ThrowsAsync<WebSocketException>(() => ReceiveAsync(clientSocket, recvSegment, cts.Token));
-                    Assert.Equal(WebSocketError.ConnectionClosedPrematurely, newReceiveException.WebSocketErrorCode);
+                
+                Assert.Equal(
+                    ResourceHelper.GetExceptionMessage("net_WebSockets_InvalidState", "Aborted", "Open, CloseSent"),
+                    newReceiveException.Message);
 
-                    Assert.Equal(WebSocketState.Open, clientSocket.State);
```

Comment:
This line pinned the incorrect behavior, and is replaced below with the correct check.